File: app/src/main/java/com/example/muslimmuhammad/kepoin/activity/BlogSingleActivity.java
package com.example.muslimmuhammad.kepoin.activity;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.muslimmuhammad.kepoin.R;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;
import com.google.firebase.storage.StreamDownloadTask;
import com.squareup.picasso.Picasso;

public class BlogSingleActivity extends AppCompatActivity {
    private String mPost_key = null;
    private DatabaseReference mDatabase;
    private ImageView mBlogSingleImage;
    private TextView mBlogSingleTitle;
    private TextView mBlogSingleDesc;
    private TextView mBlogSingleProvinsi;
    private TextView mBlogSingleKategori;
    private TextView mBlogSingleUsername;
    private Button mSingleRemoveBtn;
    private FirebaseAuth mAuth;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_blog_single);
        mPost_key = getIntent().getExtras().getString("blog_id");
        mDatabase = FirebaseDatabase.getInstance().getReference().child("Post");
        //Toast.makeText(BlogSingleActivity.this,mPost_key,Toast.LENGTH_LONG).show();
        mAuth = FirebaseAuth.getInstance();
        mBlogSingleDesc = (TextView)findViewById(R.id.single_desc);
        mBlogSingleImage = (ImageView)findViewById(R.id.single_image);
        mBlogSingleTitle = (TextView)findViewById(R.id.single_title);
        mBlogSingleKategori = (TextView)findViewById(R.id.single_kategori);
        mBlogSingleProvinsi = (TextView)findViewById(R.id.single_provinsi);
        mSingleRemoveBtn = (Button)findViewById(R.id.singleRemovebtn);
        mBlogSingleUsername = (TextView)findViewById(R.id.single_username);
        mDatabase.child(mPost_key).addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(DataSnapshot dataSnapshot) {
                String post_title = (String) dataSnapshot.child("judul").getValue();
                String post_desc = (String) dataSnapshot.child("penjelasan").getValue();
                String post_image = (String) dataSnapshot.child("image").getValue();
                String post_uid = (String)dataSnapshot.child("uid").getValue();
                String post_username = (String)dataSnapshot.child("username").getValue();
                String post_kategori = (String)dataSnapshot.child("kategori").getValue();
                String post_provinsi = (String)dataSnapshot.child("provinsi").getValue();

                mBlogSingleTitle.setText(post_title);
                mBlogSingleDesc.setText(post_desc);
                Picasso.with(BlogSingleActivity.this).load(post_image).into(mBlogSingleImage);
                mBlogSingleKategori.setText(post_kategori);
                mBlogSingleProvinsi.setText(post_provinsi);
                mBlogSingleUsername.setText(post_username);
                if (mAuth.getCurrentUser().getUid().equals(post_uid)){
                    mSingleRemoveBtn.setVisibility(View.VISIBLE);

                }
            }

            @Override
            public void onCancelled(DatabaseError databaseError) {

            }
        });
        mSingleRemoveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mDatabase.child(mPost_key).removeValue();
                Intent mainIntent = new Intent(BlogSingleActivity.this,Kepo_Post.class);
                startActivity(mainIntent);
                finish();
            }
        });
    }
}

File: app/src/main/java/com/achmad/sun3toline/kepoin/activity/BlogSingleActivity.kt
/*
 * *
 *  * Created by Achmad Fathullah on 10/3/20 9:21 PM
 *  * Copyright (c) 2020 . All rights reserved.
 *  * Last modified 10/3/20 8:20 PM
 *
 */

package com.achmad.sun3toline.kepoin.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.achmad.sun3toline.kepoin.R
import com.bumptech.glide.Glide
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*
import kotlinx.android.synthetic.main.activity_blog_single.*

class BlogSingleActivity : AppCompatActivity() {
    private var mpostKey: String = ""
    private lateinit var mDatabase: DatabaseReference
    private lateinit var mAuth: FirebaseAuth
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_blog_single)
        mpostKey = intent.extras?.getString("blog_id").toString()
        mDatabase = FirebaseDatabase.getInstance().reference.child("Post")
        mAuth = FirebaseAuth.getInstance()
        mDatabase.child(mpostKey).addValueEventListener(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                val postTitle = dataSnapshot.child("judul").value as String?
                val postDesc = dataSnapshot.child("penjelasan").value as String?
                val postImage = dataSnapshot.child("image").value as String?
                val postUid = dataSnapshot.child("uid").value as String?
                val postUsername = dataSnapshot.child("username").value as String?
                val postKategori = dataSnapshot.child("kategori").value as String?
                val postProvinsi = dataSnapshot.child("provinsi").value as String?
                single_title.text = postTitle
                single_desc.text = postDesc
                Glide.with(this@BlogSingleActivity).load(postImage).into(single_image)
                single_kategori.text = postKategori
                single_provinsi.text = postProvinsi
                single_username.text = postUsername
                if (mAuth.currentUser?.uid.toString() == postUid) {
                    singleRemovebtn.visibility = View.VISIBLE
                }
            }

            override fun onCancelled(databaseError: DatabaseError) {}
        })
        singleRemovebtn.setOnClickListener {
            mDatabase.child(mpostKey).removeValue()
            val mainIntent = Intent(this@BlogSingleActivity, PostActivity::class.java)
            startActivity(mainIntent)
            finish()
        }
    }
}